core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -129,13 +131,33 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
+        ctrl.saveBeforeUnload();
+      };
+
+      $scope.canPublish = false;
 
+      $scope.showPublishButton = function() {
         UserExplorationPermissionsService.getPermissionsAsync()
           .then(function(permissions) {
-            $scope.showPublishButton = function() {
-              return permissions.canPublish && $scope.isPrivate();
-            };
+            $scope.canPublish = permissions.canPublish;
+            $rootScope.$applyAsync();
           });
+
+        return $scope.canPublish && $scope.isPrivate();
+      };
+
+      ctrl.saveBeforeUnload = function() {
+        WindowRef.nativeWindow.addEventListener(
+          'beforeunload', ctrl.confirmBeforeLeaving);
+      };
+
+      ctrl.confirmBeforeLeaving = function(e) {
```

Comment:
use PreventReloadEventService here*

Reply:
will do that once the #11654  gets merged

Reply:
Can you please update the PR once it's merged?

Reply:
DONE